### ISSUE TITLE
Add a button to copy URL of a video to clipboard

### ISSUE THREAD

**Issue by Abanoub8** [TeamNewPipe/NewPipe#6435]
### Checklist

- [ ] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

I considered the issue, and I made a patch that adds a button, beside the open in browser, and the share, buttons.

#### Is your feature request related to a problem? Please describe it

No, my feature is not related to a problem.

#### Additional context

I have the patch and I want to know if you will be interested to accept it.

#### How will you/everyone benefit from this feature?

This will make it easier to take the URL, and save it in a file, or open it in another app, or sending it to a friend, or sync it.

**Comment by SameenAhnaf** [comment]
You can install "Copy to clipboard" app as an alternative. This app will copy link if you select the app from share menu. There are many apps just like Newpipe which don't have 'Copy link' option.

https://f-droid.org/packages/com.tengu.sharetoclipboard/

**Comment by gshirodkar** [comment]
In Android, you can also copy to clipboard from within the share menu.

**Comment by SameenAhnaf** [comment]
@gshirodkar As far as I know, only YouTube app can do so. The key reason why this issue was opened is the fact that 'Copy link' option doesn't appear in case of all apps.

**Comment by opusforlife2** [comment]
@Abanoub8 Rather than add a new button entirely, why not overload the Share button? You could make it so that long-pressing it copies the URL to the clipboard. (If you do this, make sure to also add a tooltip for taps which explain the user could long-press, like some of the other buttons.)

**Comment by infinitewaveparticle** [comment]
I like these ideas because clicking Share opens a bunch of apps that I then have to force close (this is an issue on older devices with limited CPU). Once so many apps are open simply browsing the internet will cause Android UI itself to crash and all my apps restart (which means my firewall and ad blocker are also killed).

**Comment by rey0n** [comment]
Long press on Share button to copy link is best variant.!

**Comment by gshirodkar** [comment]
@SameenAhnaf not sure if it's a feature introduced in a new version of Android, but I get the copy link for sharing from any app. 
![Screenshot_20210607-003042](https://user-images.githubusercontent.com/11176259/120959474-d704cc80-c727-11eb-8873-0b6a0adcdac4.png)

**Comment by Abanoub8** [comment]
I would say overloading the share button is the best option, it can be considered as a shortcut, and yes, it might help people with old devices and the like. And however wants can use other apps like those and copy the link as he wishes.

**Comment by SameenAhnaf** [comment]
@gshirodkar I have devices running on Android 4.4, 7.0, 10.0 and 11.0. Unfortunately, none of them show "Copy link" option in share menu. Note that Pixel devices don't have stock Android actually. Google puts a few tweaks exclusively for these devices.


@opusforlife2 @Abanoub8 I want long press on share button work for sharing URL at current time. A built-in clipboard option in share menu like "Newpipe Clipboard" option could be added. This option will copy the link directly to clipboard. Many apps like WPS Office show multiple options in share menu to perform any action directly.
![IMG_20210607_205852](https://user-images.githubusercontent.com/79245641/121039783-34098c80-c7d3-11eb-86af-f1943a236cd2.jpg)

I know, this will add an extra step. But sharing URL at current time has multiple purposes.
1. Users may want to start from a specific part.
2. Explanation or key details are marked at that part.
3. Play the selected part by other players with advanced options like zoom and pan, screenshot, subtitle up/down and a lot more.

**Comment by opusforlife2** [comment]
@gshirodkar It was introduced in Android 10.

@SameenAhnaf There is already a "Share with timestamp" button inside the player, and since the player will always be open for you to have a timestamp, then it doesn't make sense for it to be outside the player as well.

**Comment by SameenAhnaf** [comment]
@opusforlife2 Actually, I wanted to preserve the usual behaviour and avoid gestures on video context menu. Plus, it's not abnormal that a user may want to copy link directly from the player.

>@gshirodkar It was introduced in Android 10.

Not sure why but I can find it nowhere in the share menu. Could you please guide me in the chat?

**Comment by opusforlife2** [comment]
> I wanted to preserve the usual behaviour and avoid gestures on video context menu. Plus, it's not abnormal that a user may want to copy URL at the time or copy link directly from the player.

@SameenAhnaf I didn't understand that. Could you rephrase?

Regarding the copy button in the share menu, I just know it was introduced in Android 10 because I first saw it in LOS 17.1, not LOS 16. No idea what the situation is like with OEM ROMs.

**Comment by SameenAhnaf** [comment]
@opusforlife2 This is what I was calling context menu. Currently, it has no long press gesture.
<details><summary>Screenshots</summary>

![IMG_20210608_000557](https://user-images.githubusercontent.com/79245641/121069605-7e990200-c7ef-11eb-8f7f-7f4212d6a55a.jpg)

</details>

Sharing link by short press and URL at current time by long press in main player and full screen buttons are possible. Clipboard option should be added as 'Copy link' is unavailable on most devices. I prefer this option in share menu as key purpose of copying link is sharing the link to other app anyway.
>A built-in clipboard option in share menu like "Newpipe Clipboard" option could be added. This option will copy the link directly to clipboard. Many apps like WPS Office show multiple options in share menu to perform any action directly.

<details><summary>Screenshots</summary>

![IMG_20210608_001110](https://user-images.githubusercontent.com/79245641/121069026-d1be8500-c7ee-11eb-839e-2669c0f5a7cd.jpg)

</details>

**Comment by opusforlife2** [comment]
@SameenAhnaf Oh now I get it. I think you misinterpreted the "Share button" part. We're not talking about the Share option in the long-press context menu here. We're talking about the Share and Open in Browser buttons that you see when you expand the arrow next to the issue title.

**Comment by opusforlife2** [comment]
> Currently, it has no long press gesture.

@SameenAhnaf I don't think it is supposed to. I have not seen that behaviour in any app.

Regarding your second screenshot, the only thing that Share button can do right now is 1. The PR for this issue adds 3 (it's been changed. No more new button.)

And you can do 2 using the player's drop down Share button. 4 is the only one that's not available.

**Comment by SameenAhnaf** [comment]
@opusforlife2 Sorry for the misunderstanding. That's what I was referring to. Context menu is not supposed to have any long press gesture.

But you were suggesting long press on share button to work as "Copy link". How will a user be able to share with timestamp? Plus, a user may want to copy link without opening the video. How will a user be able to do that?

>You could make it so that long-pressing it copies the URL to the clipboard.

Did you see the second screenshot properly? I wish to have all four features in the same button as copying link is supposed to share (to any other user or app). Plus, a user may want to copy link of the video from context menu directly without opening the video. 

That's why, I suggested clipboard option in share menu. This is how I wish 'Share' button in video info and main player to work.
1. Short press+ Any other option: Share link to that app
2. Long press+ Any other option: Share URL at current time to that app
3. Short press+ Newpipe Clipboard: Copy link to clipboard
4. Long press+ Newpipe Clipboard: Copy URL at current time to clipboard

This is how 'Share' option in context menu should work.
1. Short press+ Any other option: Share link to that app
2. Short press+ Newpipe Clipboard: Copy link to clipboard

'Newpipe clipboard' could be renamed as 'Copy link' option as well.
